docs/extensions/attributetable.py

```diff
@@ -1,90 +1,127 @@
-from sphinx.util.docutils import SphinxDirective
-from sphinx.locale import _
-from docutils import nodes
-from sphinx import addnodes
-
-from collections import OrderedDict, namedtuple
 import importlib
 import inspect
-import os
 import re
+from typing import Any, AsyncIterable, Dict, List, NamedTuple, Optional, Tuple
+
+from docutils import nodes
+from sphinx import addnodes
+from sphinx.application import Sphinx
+from sphinx.environment import BuildEnvironment
+from sphinx.locale import _
+from sphinx.util.docutils import SphinxDirective
+from sphinx.util.typing import OptionSpec
+
+
+def returns_async_iterable(object: Any) -> bool:
+    return_type = object.__annotations__.get("return")
+    if isinstance(return_type, str):
+        module = importlib.import_module(object.__module__)
+        try:
+            return_type = getattr(module, return_type)
+        except AttributeError:  # not a global, so probably not an async iterable
+            return False
```

Comment:
This is definitely finicky and only works because incidentally a lot of the async iterables import their implementations. This might not end up being true in a future refactor.

Reply:
I understand, I'm just not entirely sure how you would avoid this unless you want to parse ast yourself.

Reply:
It'd be better to parse the docstring than parse the types.

Reply:
I'm just using inspect.isasyncgenfunction now which I think is a much better solution than either of the two propsed